cmd/snap/cmd_buy.go

```diff
@@ -90,18 +90,23 @@ func buySnap(snapName string) error {
 		return fmt.Errorf(i18n.G("cannot buy snap: it has already been bought"))
 	}
 
+	sysinfo, err := cli.SysInfo()
+	if err != nil {
+		return err
+	}
+
 	err = cli.ReadyToBuy()
 	if err != nil {
 		if e, ok := err.(*client.Error); ok {
 			switch e.Kind {
 			case client.ErrorKindNoPaymentMethods:
-				return fmt.Errorf(i18n.G(`You need to have a payment method associated with your account in order to buy a snap, please visit https://my.ubuntu.com/payment/edit to add one.
+				return fmt.Errorf(i18n.G(`You need to have a payment method associated with your account in order to buy a snap, please visit %q to add one.
 
-Once you’ve added your payment details, you just need to run 'snap buy %s' again.`), snap.Name)
+Once you’ve added your payment details, you just need to run 'snap buy %s' again.`), sysinfo.urls.paymentSetup, snap.Name)
```

Comment:
Perhaps a silly question, why are we asking snapd for the store URL and then send them back with another request?

Reply:
I'm not sure what you mean - send them back where?

Reply:
We discussed this on IRC and this was just confusion about how the patch worked.